### PR TITLE
refactor(emqx_resource): Improve grouping strategy for emqx_resource_instance

### PR DESCRIPTION
**Pull request by EMQ-YangM**

### CODE REVIEW

File: apps/emqx_resource/include/emqx_resource.hrl [hunk 1]
```diff
@@ -43,3 +43,4 @@
 -type after_query_fun() :: {fun((...) -> ok), Args :: [term()]}.
 
 -define(TEST_ID_PREFIX, "_test_:").
+-define(DEFAULT_RESOURCE_GROUP, <<"default">>).
```
Comment (terry-xiaoyu) [inline apps/emqx_resource/include/emqx_resource.hrl:46]: If the only purpose of this macro is for testing, we can defined it in the test cases.

Comment (EMQ-YangM) [reply apps/emqx_resource/include/emqx_resource.hrl:46]: I will move it to the test file.

File: apps/emqx_bridge/src/emqx_bridge.erl [hunk 2]
```diff
@@ -222,7 +222,8 @@ create(BridgeId, Conf) ->
 create(Type, Name, Conf) ->
     ?SLOG(info, #{msg => "create bridge", type => Type, name => Name,
         config => Conf}),
-    case emqx_resource:create_local(resource_id(Type, Name), emqx_bridge:resource_type(Type),
+    Group = maps:get(group, Conf, <<"default">>),
```
Comment (terry-xiaoyu) [inline apps/emqx_bridge/src/emqx_bridge.erl:225]: We should hard code the group name to `<<"emqx_bridge">>`

File: apps/emqx_retainer/src/emqx_retainer.erl [hunk 3]
```diff
@@ -480,6 +480,7 @@ create_resource(Context, #{type := DB} = Config) ->
     ResourceID = erlang:iolist_to_binary([io_lib:format("~ts_~ts", [?APP, DB])]),
     case emqx_resource:create(
            ResourceID,
+           <<"default">>,
```
Comment (terry-xiaoyu) [inline apps/emqx_retainer/src/emqx_retainer.erl:483]: `<<"emqx_retainer">>`

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**